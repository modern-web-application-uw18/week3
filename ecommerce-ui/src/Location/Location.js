import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Location extends Component {
    render() {
        return ( <span className="Bolded"> {this.props.location.city}, {this.props.location.country} </span>
        )};
}

Location.propTypes = {
    city: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired
}

Location.defaultProps = {
    city: 'Unknown',
    country: 'Unknown'
}

export default Location;