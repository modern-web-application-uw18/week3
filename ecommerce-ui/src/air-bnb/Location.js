import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Location extends Component {

    render() {

        const {
            city,
            country
        } = this.props.location;

        return (
            <span>{city}, {country}</span>
        );
    }
}

Location.propTypes = {
    city: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired
};

Location.defaultProps = {
    city: 'Unknown',
    country: 'Unknown'
};

export default Location;
