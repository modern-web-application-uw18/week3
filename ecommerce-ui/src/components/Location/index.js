import React, { Component } from 'react'

class Location extends Component {

  render() {
    const{
      title,
      houseType,
      image,
      location,
      payment,
      host,
      rating 
    } = this.props.airbandbs;
    
    console.log(this.props.airbandbs)
    
    return(

      <div className="location">
        <img src={image} alt={title} />
        <h2>{title} </h2>
        <p>{houseType} â€¢ {location.city}</p>
        <p>Host: {host.name} <br /> </p>
        <button className="btn">Add to Cart</button>
      </div>
    );
  }
}

export default Location;