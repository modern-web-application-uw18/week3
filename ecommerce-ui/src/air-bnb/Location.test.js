import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Location from './Location';

describe('Location', () => {
    it('renders without crashing', () => {
        const location = {"city": "Miami", "country": "USA"};
        const div = document.createElement('div');
        ReactDOM.render(<Location location={location} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with city and country', () => {
        const location = {"city": "Helsinki", "country": "Finland"};
        const component = TestRenderer.create(
            <Location location={location} />);
            expect(component.toJSON()).toMatchSnapshot();
    });
});