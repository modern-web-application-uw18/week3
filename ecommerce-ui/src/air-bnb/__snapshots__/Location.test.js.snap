// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location should render with city and country 1`] = `
<span>
  Helsinki
  , 
  Finland
</span>
`;
